packages/flat-components/src/components/SaveAnnotationModal/index.tsx

```diff
@@ -0,0 +1,128 @@
+import "./style.less";
+
+import pLimit from "p-limit";
+import classNames from "classnames";
+import React, { useCallback, useEffect, useMemo, useRef, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { Modal, message } from "antd";
+
+function download(
+    canvas: HTMLCanvasElement,
+    filename = "annotation.png",
+    failText = "Save image failed",
+): void {
+    try {
+        const a = document.createElement("a");
+        a.download = filename;
+        a.href = canvas.toDataURL();
+        a.style.display = "none";
+        document.body.appendChild(a);
+        a.click();
+        document.body.removeChild(a);
+    } catch (err) {
+        console.error(err);
+        message.error(failText);
+    }
+}
+
+export interface SaveAnnotationModalProps {
+    visible: boolean;
+    onClose: () => void;
+    images: Array<() => Promise<HTMLCanvasElement | null>>;
+}
+
+export const SaveAnnotationModal: React.FC<SaveAnnotationModalProps> = ({
+    visible,
+    onClose,
+    images,
+}) => {
+    const { t } = useTranslation();
+
+    const params = useMemo(() => {
+        const limit = pLimit(1);
+        return images.map(image => limit(image));
```

Comment:
```suggestion
        return images.map(limit);
```

Reply:
`limit` has such type signature:

```ts
limit(fn, ...args)
```

I don't think it is right to pass map arguments (index, array) to the function.

Reply:
Okey. You are right.